### PR TITLE
Detect integers 6644 v10

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6644 and all subtickets
https://redmine.openinfosecfoundation.org/issues/6645
https://redmine.openinfosecfoundation.org/issues/6646
https://redmine.openinfosecfoundation.org/issues/6647
https://redmine.openinfosecfoundation.org/issues/6648
https://redmine.openinfosecfoundation.org/issues/6628

Describe changes:
- detect/integers: support hexadecimal notation for parsing
- detect/integers: add mode for negated range
- detect/integers: rust derive for enumerations
- detect/integers: keywords now accept bitmasks
- doc: detect/integers

https://github.com/OISF/suricata/pull/101759 with 
- clearer doc
- parsing error on `&0xc0=12` as it could never match

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.

ERROR: QA failed on SURI_TLPW1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .flow.spare | 1961331 | 2161217 | 110.19% |
| | SURI_TLPW1_stats_chk | | |
| .uptime | 137 | 144 | 105.11% | 

Pipeline 17622

File: doc/userguide/rules/integer-keywords.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* Inequality
+* Greater than
+* Less than
+* Range
+* Negated range
+* Bitmask
+* Negated Bitmask
+
+Comparisons are strict by default.
+That means a range between 1 and 4 will match 2 and 3, but neither 1 nor 4.
```
Comment (jufajardini) [inline doc/userguide/rules/integer-keywords.rst:33]: This connects to @jlucovsky question here, right? https://github.com/OISF/suricata/pull/10208#discussion_r1460916496

I think it's good to make this into a note or something that is highlighted a bit more.

File: doc/userguide/rules/payload-keywords.rst [hunk 2]
```diff
@@ -280,6 +280,8 @@ bsize
 With the ``bsize`` keyword, you can match on the length of the buffer. This adds
 precision to the content match, previously this could have been done with ``isdataat``.
 
+bsize uses an :ref:`unsigned 64-bits integer <rules-integer-keywords>`.
+
 An optional operator can be specified; if no operator is present, the operator will
 default to '='. When a relational operator is used, e.g., '<', '>' or '<>' (range),
 the bsize value will be compared using the relational operator. Ranges are inclusive.
```
Comment (jufajardini) [inline doc/userguide/rules/payload-keywords.rst:287]: Question: considering Jeff's question about range being inclusive or exclusive, and this note here while the new description says ranges are exclusive, are we changing behavior, and do we need to adjust behavior, or just documentation?

**Review by jufajardini — COMMENTED** [review]
Comments by Jeff on another PR #10208 made me ponder about changes seen here. Left inline comments about it.

File: doc/userguide/rules/integer-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/integer-keywords.rst:33]: >This connects to @jlucovsky question here, right? https://github.com/OISF/suricata/pull/10208#discussion_r1460916496

Perfectly right.

How do I make a note ?

File: doc/userguide/rules/payload-keywords.rst (hunk 2 above)
Comment (catenacyber) [reply doc/userguide/rules/payload-keywords.rst:287]: We are not changing behavior. Ranges are exclusive, and more generally speaking comparisons are strict, except for nfs.procedure and nfs.version which have this historic behavior...

File: doc/userguide/rules/integer-keywords.rst (hunk 1 above)
Comment (jufajardini) [inline doc/userguide/rules/integer-keywords.rst:33]:
```suggestion
.. note::

    Comparisons are strict by default.
    That means a range between 1 and 4 will match 2 and 3, but neither 1 nor 4.
```

Comment (jufajardini) [reply doc/userguide/rules/integer-keywords.rst:33]: Just added a suggestion that shows how to do it :)

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10222

File: doc/userguide/rules/payload-keywords.rst (hunk 2 above)
Comment (jufajardini) [reply doc/userguide/rules/payload-keywords.rst:287]: Oh, right... So was this bit of explanation wrong?

Comment (catenacyber) [reply doc/userguide/rules/payload-keywords.rst:287]: Yes, this doc is wrong (it does not reflect the behavior cf uint.rs
```
        DetectUintMode::DetectUintModeRange => {
            if val > x.arg1 && val < x.arg2 {
                return true;
            }
        }
```

File: doc/userguide/rules/integer-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/integer-keywords.rst:33]: ok thanks

File: doc/userguide/rules/payload-keywords.rst (hunk 2 above)
Comment (jufajardini) [reply doc/userguide/rules/payload-keywords.rst:287]: ooops 😅

**Pull request closed without merging**